From c60cc9983c9bbc0c16543296d37c792a431b7c39 Mon Sep 17 00:00:00 2001
Message-ID: <c60cc9983c9bbc0c16543296d37c792a431b7c39.1710967273.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1710967273.git.sharlatanus@gmail.com>
References: <cover.1710967273.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sun, 10 Mar 2024 08:49:57 +0000
Subject: [PATCH 03/49] gnu: python-blosc: Move to python-compression.

* gnu/packages/python-xyz.scm (python-blosc): Move from here ...
* gnu/packages/python-compression.scm: ... to here.

Change-Id: Ieed1b399602d0ba64b065d1c03ef0855ac52f71e
---
 gnu/packages/python-compression.scm | 47 ++++++++++++++++++++++++++++-
 gnu/packages/python-xyz.scm         | 44 ---------------------------
 2 files changed, 46 insertions(+), 45 deletions(-)

diff --git a/gnu/packages/python-compression.scm b/gnu/packages/python-compression.scm
index 691fbd3065..ef289bc4d1 100644
--- a/gnu/packages/python-compression.scm
+++ b/gnu/packages/python-compression.scm
@@ -3,7 +3,7 @@
 ;;; Copyright © 2017, 2019, 2021, 2022 Ricardo Wurmus <rekado@elephly.net>
 ;;; Copyright © 2017 Nikita <nikita@n0.is>
 ;;; Copyright © 2017 Julien Lepiller <julien@lepiller.eu>
-;;; Copyright © 2018-2020, 2022 Efraim Flashner <efraim@flashner.co.il>
+;;; Copyright © 2018-2020, 2022, 2023 Efraim Flashner <efraim@flashner.co.il>
 ;;; Copyright © 2020 Nicolas Goaziou <mail@nicolasgoaziou.fr>
 ;;; Copyright © 2020, 2022, 2023 Marius Bakke <marius@gnu.org>
 ;;; Copyright © 2021 Brendan Tildesley <mail@brendan.scot>
@@ -38,6 +38,7 @@ (define-module (gnu packages python-compression)
   #:use-module (gnu packages libffi)
   #:use-module (gnu packages compression)
   #:use-module (gnu packages crates-io)
+  #:use-module (gnu packages cmake)
   #:use-module (gnu packages check)
   #:use-module (gnu packages maths)
   #:use-module (gnu packages pkg-config)
@@ -49,6 +50,50 @@ (define-module (gnu packages python-compression)
   #:use-module (gnu packages rust-apps)
   #:use-module (gnu packages sphinx))
 
+(define-public python-blosc
+  (package
+    (name "python-blosc")
+    (version "1.11.1")
+    (source
+     (origin
+       (method url-fetch)
+       (uri (pypi-uri "blosc" version))
+       (sha256
+        (base32
+         "0xmjs28sgpnb940zrhw010dq2m9d8a5h4fgnjyk6645fgfr1j8f2"))
+       (snippet
+        #~(begin (use-modules (guix build utils))
+                 (delete-file-recursively "blosc/c-blosc")))))
+    (build-system python-build-system)
+    (arguments
+     (list #:phases
+           #~(modify-phases %standard-phases
+               (add-after 'unpack 'find-blosc
+                 (lambda* (#:key inputs #:allow-other-keys)
+                   (setenv "USE_SYSTEM_BLOSC" "1")
+                   (setenv "Blosc_ROOT" #$(this-package-input "c-blosc"))))
+               (replace 'check
+                 (lambda* (#:key tests? #:allow-other-keys)
+                   (when tests?
+                     (invoke "python" "-m" "blosc.test")))))))
+    (propagated-inputs
+     (list python-scikit-build python-numpy))
+    (inputs (list c-blosc))
+    (native-inputs (list cmake-minimal))
+    (home-page "https://github.com/blosc/python-blosc")
+    (synopsis "Python wrapper for the Blosc data compressor library")
+    (description "Blosc is a high performance compressor optimized for binary
+data.  It has been designed to transmit data to the processor cache faster
+than the traditional, non-compressed, direct memory fetch approach via a
+@code{memcpy()} system call.
+
+Blosc works well for compressing numerical arrays that contains data with
+relatively low entropy, like sparse data, time series, grids with
+regular-spaced values, etc.
+
+This Python package wraps the Blosc library.")
+    (license license:bsd-3)))
+
 (define-public python-multivolumefile
   (package
     (name "python-multivolumefile")
diff --git a/gnu/packages/python-xyz.scm b/gnu/packages/python-xyz.scm
index 7ed8c945be..7eacce5caf 100644
--- a/gnu/packages/python-xyz.scm
+++ b/gnu/packages/python-xyz.scm
@@ -27394,50 +27394,6 @@ (define-public python-locket
 they use the same path.")
     (license license:bsd-2)))
 
-(define-public python-blosc
-  (package
-    (name "python-blosc")
-    (version "1.11.1")
-    (source
-     (origin
-       (method url-fetch)
-       (uri (pypi-uri "blosc" version))
-       (sha256
-        (base32
-         "0xmjs28sgpnb940zrhw010dq2m9d8a5h4fgnjyk6645fgfr1j8f2"))
-       (snippet
-        #~(begin (use-modules (guix build utils))
-                 (delete-file-recursively "blosc/c-blosc")))))
-    (build-system python-build-system)
-    (arguments
-     (list #:phases
-           #~(modify-phases %standard-phases
-               (add-after 'unpack 'find-blosc
-                 (lambda* (#:key inputs #:allow-other-keys)
-                   (setenv "USE_SYSTEM_BLOSC" "1")
-                   (setenv "Blosc_ROOT" #$(this-package-input "c-blosc"))))
-               (replace 'check
-                 (lambda* (#:key tests? #:allow-other-keys)
-                   (when tests?
-                     (invoke "python" "-m" "blosc.test")))))))
-    (propagated-inputs
-     (list python-scikit-build python-numpy))
-    (inputs (list c-blosc))
-    (native-inputs (list cmake-minimal))
-    (home-page "https://github.com/blosc/python-blosc")
-    (synopsis "Python wrapper for the Blosc data compressor library")
-    (description "Blosc is a high performance compressor optimized for binary
-data.  It has been designed to transmit data to the processor cache faster
-than the traditional, non-compressed, direct memory fetch approach via a
-@code{memcpy()} system call.
-
-Blosc works well for compressing numerical arrays that contains data with
-relatively low entropy, like sparse data, time series, grids with
-regular-spaced values, etc.
-
-This Python package wraps the Blosc library.")
-    (license license:bsd-3)))
-
 (define-public python-partd
   (package
     (name "python-partd")
-- 
2.41.0

